File: legacy/OrderMatchingEngine/src/main/java/com/ndgndg91/ordermatchingengine/global/OrderServiceAdviser.java
package com.ndgndg91.ordermatchingengine.global;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
public class OrderServiceAdviser {

    @ExceptionHandler(OrderServiceException.class)
    public ResponseEntity<ApiError> orderServiceException(OrderServiceException e) {
        return ResponseEntity
                .status(e.getStatus())
                .body(new ApiError(e.getStatus(), e.getMessage(), e.getPath(), e.getTimestamp()));
    }
}

File: OrderMatchingEngineKotlin/src/main/kotlin/com/ndgndg91/ordermatchingenginekotlin/global/OrderServiceAdviser.kt
package com.ndgndg91.ordermatchingenginekotlin.global

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class OrderServiceAdviser {

    @ExceptionHandler(OrderServiceException::class)
    fun orderServiceException(e: OrderServiceException): ResponseEntity<ApiError> {
       return ResponseEntity.status(e.status)
           .body(e.message?.let { ApiError(e.status, it, e.path, e.timestamp) });
    }
}